src/Products/ZCatalog/query.py

```diff
@@ -88,8 +88,7 @@ def __init__(self, request, iid, options=(), operators=('or', 'and'),
 
             for field in request.keys():
                 if field.startswith(iid + '_'):
-                    iid_tmp, op = field.split('_')
-
+                    tmp, op = field.split(iid + '_')
```

Comment:
`op = field[len(iid)+1:]` seems more natural and avoids an unused variable plus protects you from stumbling in the same problem:
```
>>> iid = "a"
>>> field = "a_a_a"
>>> tmp, op = field.split(iid + '_')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: too many values to unpack (expected 2)
```
This will work as well:
```
field.partition(iid + '_')[-1]
```

Reply:
nice catch ... I will change to last suggestion